### PR TITLE
Jetpack Mobile: Scan UiStates part(ii)

### PR DESCRIPTION
**Pull request by ashiagr**
Parent https://github.com/wordpress-mobile/WordPress-Android/issues/13326

PS: Size of the PR has increased because of new xml layouts, extracting of styles to a new class. I can split tests (last commit) into a separate PR if it helps in reviewing this PR.

### Description

This PR is a step forward in refining `UiStates` for the `Scan` screen. It mainly includes changes to below classes:

`ScanListItemState` updated to include
- `ScanState` - `ButtonAction` (FixAll, Scan) - (only display logic) do not perform any action yet.
- `ThreatItemState` (corresponding `ThreatViewHolder`, layout etc added)

`ScanStateListItemBuilder`
- Mapping for `ScanState` extracted  in the new `ScanStateListItemBuilder` class, dynamic texts added 

`ScanViewModel`
- `Content` `UiState` updated to include `ScanState` buttons and threats

There will be more states (e.g. after fix action) later, this PR covers only below states, dynamic texts, buttons (displayed when screen is shown) and corresponding tests.
 
 ScanStateModel.State | SCANNING | IDLE | IDLE
-- | -- | -- | --
ScanListItemState  | Scanning | ThreatsNotFound | ThreatsFound (none, some or all fixable)
icon | ic_scan_scanning | ic_scan_idle_threats_not_found | ic_scan_idle_threats_found
title | Preparing to scan | Don\'t worry about a thing | Your site may be at risk
description | Welcome to Jetpack Scan, we are taking a first look at your site now and the result will be with you soon. \n \n We will send you a notification once the scan is complete. In the meantime feel free to continue to use your site as normal. You can check the progress at anytime. | The last Jetpack scan ran _{no. of hours/mins ago or in a few seconds}_ and everything looked great. Run a manual scan now or wait for Jetpack to scan your site later today. | The scan found {no. of threats} potential threats with _{site name or this site}_. Please review them below and take action. We\'re _{here to help}_ if you need us.
fix all button | - | - | Fix all (visibility based on fixable status)
scan button | - | Scan now | Scan again

### To Test

Prerequisite: Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.

- Select a site with scan capability on the My Site tab
- Notice that scan menu is displayed
- Click scan menu item
- Notice that icons, (dynamic) texts,  (dynamic) buttons display as per the table above corresponding to the mentioned states.
- Notice that threats have started displaying (title, description based on threat type yet to be updated)

### Notes

1. Design, icons not final. Ignore all styles.
2. `ScanStatusService` uses `FluxC.ThreatStore` to access threats (only basic details) temporarily just to start displaying the threats, threats count. It needs to be updated later on and so haven't included tests for `ScanStatusService`.
3. "here to help" in the `ThreatsFound` state is not highlighted as a link.

### Merge Instructions

1. Make sure #13578 is merged
2. Ping me for merge. We need to remove fluxc commit hash for getting threats temporarily.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanStateListItemBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    )
+
+    private fun buildFixAllButtonAction(onFixAllButtonClicked: () -> Unit) = ButtonAction(
+        title = UiStringRes(R.string.threats_fix_all),
+        visibility = true,
+        onClicked = onFixAllButtonClicked
+    )
+
+    private fun buildLastScanDescription(timeInMs: Long): UiStringResWithParams {
+        val durationInMs = System.currentTimeMillis() - timeInMs
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanStateListItemBuilder.kt]: 🔍 Nitpick: We might want to use injectable DateProvider here to make our tests deterministic. We might also want to add tests for the "scan_in_hour, ..." placeholders. If you plan to add those in future PRs feel free to ignore this.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/adapters/viewholders/ScanStateViewHolder.kt [hunk 2]
```diff
@@ -1,23 +1,40 @@
...
+            scan_state_description.text = getTextOfUiString(context, scanState.scanDescription)
         }
     }
+
+    private fun bindButtonAction(
+        button: MaterialButton,
+        action: ButtonAction?
+    ) {
+        action?.let {
+            button.text = uiHelpers.getTextOfUiString(itemView.context, action.title)
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/adapters/viewholders/ScanStateViewHolder.kt]: 🔍 Nitpick: We might want to consider using "uiHelpers.setTextOrHide(..)" and drop the "action?.let". I usually try to have as little nesting in ViewHolders as possible to make sure someone doesn't add a line into one branch and forgets to add it into the other branch (which might cause issues during recycling). Having said that, it's very much about personal preference.

```
uiHelpers.setTextOrHide(button, action?.title)
button.setOnClickListener { action?.onClicked.invoke() }
```

**Review by malinajirka — APPROVED** [review]
Thanks @ashiagr! Really great job :). I left to very nitpicky comments, the code looks great. Feel free to merge this PR.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/adapters/viewholders/ScanStateViewHolder.kt (hunk 2 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/adapters/viewholders/ScanStateViewHolder.kt]: In 9cac9e5

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanStateListItemBuilder.kt (hunk 1 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanStateListItemBuilder.kt]: In 1769684

**Comment by ashiagr** [comment]
Thanks for the review @malinajirka ! I've addressed your comments. Ready for another look 🙏.

PS: Don't merge this PR yet. I have to integrate FluxC threats updates and remove temporary FluxC commit hash from this PR.

**Review by malinajirka — APPROVED** [review]
Thanks for the changes ;)! LGTM

**Pull request merged**